src/components/Popover/Popover.js

```diff
@@ -60,6 +65,12 @@ class Popover extends Component {
     document.addEventListener('click', this.handleDocumentClick, false);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.isOpen && !this.state.isOpen) {
+      this.props.onClose();
+    }
+  }
+
```

Comment:
Interesting! What's your reason to use `componentDidUpdate` instead of just calling `this.props.onClose` inside `this.closePopover`? 🙂

Reply:
Ah, is it because of `this.handleReferenceClick` where you're toggling the open state?

Reply:
@connor-baer the popover has no onClose element reference. I guess this is why.

Reply:
Yea. I was having the same code at two places and it was getting hard to read/follow. Next iteration could possibly add a third one. 

I am not sure about this pattern yet, but I've seen it at some places and it seems really concise.